clang/Cursor.d

```diff
@@ -18,28 +22,54 @@ struct Cursor
 {
     mixin CX;
 
+    private static const CXCursorKind[string] predefined;
+
+    static this()
+    {
+        predefined = queryPredefined();
+    }
+
     @property static Cursor empty ()
     {
         auto r = clang_getNullCursor();
         return Cursor(r);
     }
 
-    @property string spelling ()
+    @property string spelling () const
     {
         return toD(clang_getCursorSpelling(cx));
     }
 
-    @property CXCursorKind kind ()
+    @property CXCursorKind kind () const
     {
         return clang_getCursorKind(cx);
     }
 
-    @property SourceLocation location ()
+    @property SourceLocation location () const
     {
         return SourceLocation(clang_getCursorLocation(cx));
     }
 
-    @property Type type ()
+    @property TokenRange tokens() const
+    {
+        import std.algorithm.mutation : stripRight;
+
+        CXTranslationUnit unit = clang_Cursor_getTranslationUnit(cx);
+        CXToken* tokens = null;
+        uint numTokens = 0;
+        clang_tokenize(unit, extent.cx, &tokens, &numTokens);
+        auto result = TokenRange(unit, tokens, numTokens);
+
+        // For some reason libclang returns some tokens out of cursors extent.cursor
+        return stripRight!((token) => !intersects(extent, token.extent))(result);
```

Comment:
* UFCS is still not used
* It's not necessary have parentheses around the parameter for a lambda if it only has one parameter